File: src/Task.ts
import { TaskStatus, TASK_STATUS } from './constants'

export class Task {
  id: number
  description: string
  status: TaskStatus
  createdAt: Date
  updatedAt?: Date

  constructor(description: string) {
    this.description = description
    this.status = TASK_STATUS.TODO
    this.createdAt = new Date()
    this.id = 0
  }
}

File: cpp/src/Task.cpp
#include "Task.hpp"

Task::Task()
    : id(0), status_code(StatusCode::TODO), description(""), created_at(""),
      updated_at("") {};

Task::Task(std::string description)
    : id(0), status_code(StatusCode::TODO), description(description),
      created_at(""), updated_at("") {};